sstable: improve descriptions of capped.*deletion_time

tchaikov opened the pull request:
before this change, they reads

> Was local deletion time capped at ...

and

> Was partition tombstone deletion time capped at ...

the "Was" part is confusing. and the first description is not accurate enough. so let's improve them a little bit.

tchaikov commented:
@bhalevy hi Benny, do you mind taking a look?

bhalevy commented:
@annastuchlik please review the phrasing, I'm not sure how to describe those metrics in the most simple terms.

scylladb-promoter commented:
CI state **`SUCCESS`** - https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/3164/

annastuchlik commented:
@bhalevy @tchaikov I'm sorry for not replying sooner; I was out of the office.

To suggest a description, I need to understand what the metrics mean.
My current understanding is the local deletion time is somehow capped, and the metrics show the number of tombstones whose local deletion time reached that maximum limit. Is that correct?

If that's the case, we could improve the clarity by saying:

```
Number of SStables with tombstones whose local deletion time reached the maximum allowed value in Statistics
Number of tombstones whose local deletion time reached the maximum allowed value
```

tchaikov commented:
> @bhalevy @tchaikov I'm sorry for not replying sooner; I was out of the office.
> 
> To suggest a description, I need to understand what the metrics mean. My current understanding is the local deletion time is somehow capped, and the metrics show the number of tombstones whose local deletion time reached that maximum limit. Is that correct?

there are two metrics here.

probably i can provide more context here. we write sstables to disk to persist the data which are still alive, and also the tombstones which shadow previous alive data. the alive data and tombstones are called mutations. the sstables are like chunks of mutations streams. so a sstable can contains multiple tombstones. and the `local_deletion_time` is a property of a certain tombstone. it is capped when its on-disk representation (32-bit) is not able to encode its value (64-bit internally). this is considered a dangerous signal. so, when we write a new sstable to disk, we check if the sstable contains any tombstone whose local_deletion_time got capped before. if the answer is "yes", `capped_local_deletion_time` is incremented.

but whenever we cap a tombstone, `capped_tombstone_deletion_time` is incremented. i think you are referencing this one.

both metrics are measured on a per-table basis.

> 
> If that's the case, we could improve the clarity by saying:
> 
> ```
> Number of SStables with tombstones whose local deletion time reached the maximum allowed value in Statistics
> Number of tombstones whose local deletion time reached the maximum allowed value
> ```

i see you are replacing "capped" with "reached". i feel that it is a degraded form of "capped" in two ways:

- "capped" might not be accurate enough. but it implies that
  1. the value exceeds a limit
  2. we enforce the limit on it
- the mathematics translation of "reaches" is "==", not ">"

i will change to "reached", if you still think it's better.

annastuchlik commented:
Thank you for the explanation. It's very useful - not only in the context of this PR.
I agree that "reach" may not be sufficient to imply that the limit is enforced. I'd leave "capped," but use it in the passive:

```
Number of SStables with tombstones whose local deletion time was capped at the maximum allowed value in Statistics
Number of tombstones whose local deletion time was capped at the maximum allowed value
```

tchaikov commented:
v2:

- incorporated Anna's suggestion.

@annastuchlik hi Anna, thank you! i incorporated your suggestion in the latest revision. could you take another look?

annastuchlik reviewed (approved):
LGTM

scylladb-promoter commented:
CI state **`SUCCESS`** - https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/3556/

tchaikov commented:
@scylladb/scylla-maint hello maintainers, could you help queue this change?
